### DISCUSSION TITLE
Application-specific terminals

### DISCUSSION THREAD

**Discussion by cweagans** [ghostty-org/ghostty#3821] [Feature Requests, Ideas] [answered]
I've been kicking around ideas for how to implement https://www.cweagans.net/2023/09/application-specific-terminals/ with Ghostty -- I think I might be able to do it with separate config files that I supply as a CLI flag, but I'm wondering if maybe there'd be some interest in making it so that Ghostty can export application specific terminals as a core feature? Really, I just want to be able to Cmd+Tab to a Neovim window.

**Comment by pluiedev** [comment, marked as answer]
Personally IMO this is too niche as a core feature. Either the commandline flag, a future plugins API or #2353 sounds like a better fit.

**Reply by cweagans** [reply]
I don't think it makes sense to add a _feature_ for it per se, but @mitchellh asked me to open a discussion when I asked about it in Discord 🤷🏻‍♂️

**Reply by jcollie** [reply]
It's easy enough to do this on Linux with a custom `.desktop` file and some custom config files. On macOS this would be much more difficult due to the way macOS launches applications. So really this comes down to being able to open a new window with a custom set of configs so this probably is better as a design criteria for the future API or plugin system rather than a core feature.

**Reply by cweagans** [reply]
Yep! I wrote at length about how to get macOS to play nice, but I did it with wezterm (link above in the topic). Being able to include an _extra_ config that gets layered on top of the normal config would be ideal. That avoids needing conditionals in the config + inspecting env vars from that config.
